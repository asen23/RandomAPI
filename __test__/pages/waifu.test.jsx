import { render, screen } from "@testing-library/react"
import userEvent from "@testing-library/user-event"
import Waifu from "pages/waifu/index"

const dummyWaifu = [{ url: "file1.jpg" }, { url: "file2.jpg" }]

describe("Waifu", () => {
    beforeEach(() => {
        fetch.resetMocks()
    })

    it("renders unchanged", async () => {
        fetch.mockResponseOnce(JSON.stringify(dummyWaifu[0]))

        const { asFragment } = render(<Waifu />)
        expect(asFragment()).toMatchSnapshot()

        await screen.findByAltText("waifu")

        expect(asFragment()).toMatchSnapshot()
    })

    it("gives new waifu", async () => {
        fetch.mockResponses(...dummyWaifu.map((e) => JSON.stringify(e)))

        render(<Waifu />)

        const firstImage = await screen.findByAltText("waifu")
        expect(firstImage.src).toContain(dummyWaifu[0].url)

        const button = screen.getByRole("button")
        userEvent.click(button)

        const loading = await screen.findByText("Loading...")
        expect(loading).toBeDefined()

        const secondImage = await screen.findByAltText("waifu")
        expect(secondImage.src).toContain(dummyWaifu[1].url)
    })
})
