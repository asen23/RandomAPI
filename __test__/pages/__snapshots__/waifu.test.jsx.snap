// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Waifu renders unchanged 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Waifu
    </h1>
    <p>
      Loading...
    </p>
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random waifu
    </button>
  </div>
</DocumentFragment>
`;

exports[`Waifu renders unchanged 2`] = `
<DocumentFragment>
  <div
    class="flex justify-center items-center flex-col min-h-screen gap-8 bg-primary"
  >
    <h1
      class="text-4xl font-bold"
    >
      Random Waifu
    </h1>
    <img
      alt="waifu"
      class="max-w-[80%] max-h-[60vh]"
      src="file1.jpg"
    />
    <button
      class="border border-gray-400 px-2 py-1 rounded-lg hover:bg-white"
    >
      Get random waifu
    </button>
  </div>
</DocumentFragment>
`;
